src/lib/Libnet/hnls.c

```diff
@@ -574,53 +574,162 @@ free_if_info(struct log_net_info *ni)
 
 /**
 * @brief
-	Get a list of all IPs
+	Get a list of all IPs (ipv4) for a given hostname
+*
+* @return
+*	Comma separated list of IPs in string format
 *
-* @return 
-*	Comma separated list of ips in string format
-* 
 * @par Side Effects:
 *	None
 *
 * @par MT-safe: Yes
 *
-* @param[out]   msg - error message returned if system calls not successful
+* @param[in]	host        - hostname of the current host to resolve IPs for
+* @param[out]   msg_buf     - error message returned if system calls not successful
+* @param[in]    msg_buf_len - length of the message buffer passed
+*
+*/
+static char *
+get_host_ips(char *host, char *msg_buf, size_t msg_buf_len)
+{
+	struct addrinfo *aip, *pai;
+	struct addrinfo hints;
+	int rc = 0;
+	char buf[NETADDR_BUF] = {'\0'};
+	int count = 0;
+	char *nodenames = NULL;
+	char *tmp, *p;
+	int len, hlen;
+
+	errno = 0;
+
+	memset(&hints, 0, sizeof(struct addrinfo));
+	hints.ai_family = AF_INET;
+	hints.ai_socktype = SOCK_STREAM;
+	hints.ai_protocol = IPPROTO_TCP;
+
+	if ((rc = getaddrinfo(host, NULL, &hints, &pai)) != 0) {
+		snprintf(msg_buf, msg_buf_len, "Error %d resolving %s\n", rc, host);
+		return NULL;
+	}
+
+	len = 0;
+	count = 0;
+	for (aip = pai; aip != NULL; aip = aip->ai_next) {
+		if (aip->ai_family == AF_INET) { /* for now only count IPv4 addresses */
+			struct sockaddr_in *sa = (struct sockaddr_in *) aip->ai_addr;
+			if (ntohl(sa->sin_addr.s_addr) >> 24 == IN_LOOPBACKNET)
+				continue;
+			sprintf(buf, "%s", netaddr(sa));
```

Comment:
Not here. We should use snprintf where we do not have a handle on how much data would be written, and if there are no problems expected from a truncation of the data. The best example of this is an error message, where i have used it. 

For this usage, which happens to be an IP address, it serves no purpose to use snprintf. A truncated (but not crashing code) would eventually crash since the ip address is not fully formed. Even if i check the return type of snprintf, there is nothing else i can do except increase the destination buffer size (which is a program time check - not runtime). 

In this case we have ensure that the destination buffer is larger, and in this case it is 80 chars, which is more than enough.